pkg/volume/scaleio/sio_client.go

```diff
@@ -337,13 +338,23 @@ func (c *sioClient) getGuid() (string, error) {
 // emc-vol-<mdmID-volID> e.g.:
 // emc-vol-788d9efb0a8f20cb-a2b8419300000000
 func (c *sioClient) getSioDiskPaths() ([]os.FileInfo, error) {
+	// first check if sioDiskIDPath exists before trying to read it
+	sioDiskIDPathExists, err := volumeutil.PathExists(sioDiskIDPath)
+	if err != nil {
+		glog.Error(log("failed to check existence of %s: %v", sioDiskIDPath, err))
+		return nil, err
+	}
+	result := []os.FileInfo{}
+	if !sioDiskIDPathExists {
+		return result, nil
+	}
+	// proceed to read sioDiskIDPath since we know it exists
 	files, err := ioutil.ReadDir(sioDiskIDPath)
 	if err != nil {
```

Comment:
Can you just check `ReadDir` err here?

```go
files, err := ioutil.ReadDir(sioDiskIDPath)
if err != nil {
    if !os.IsNotExists(err) {
        return []os.FileInfo{}, nil
    } else {
        return nil, err
}
```

Reply:
Sure ... I have changed the PR to follow above.